### PR TITLE
Added resource state checker which enforces resource states

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- makes sure that the value of the state field of resources is either a particular value, or within an array of values
- default value is "translated"

### CODE REVIEW

**Review by gouniLee — APPROVED** [review]
Approved.
All the test cases are passed on QT v6.4.0 as well.

**Comment by gouniLee** [comment]
> Approved. All the test cases are passed on QT v6.4.0 as well.

my previous comment was wrong. it shouldn't be here. so I set it to hidden. sorry for the confusion.

File: test/testRules.js [hunk 1]
```diff
@@ -996,6 +997,192 @@ export const testRules = {
...
+            }),
+            file: "x/y"
+        });
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceStateCheckerMatchDefaultError: function(test) {
+        test.expect(2);
```
Comment (gouniLee) [inline test/testRules.js:1156]: Don't we need to add a test case with testing the actual xliff file?

Comment (ehoogerbeets) [reply test/testRules.js:1156]: These tests and the others in this file assume that that the resource file is converted correctly into Resource instances. If so, then this Rule instance will work too.

**Review by gouniLee — APPROVED** [review]

**Pull request merged**